File: Algorithm and Data Structure/Baekjoon/12001번 Load Balancing (Bronze)/boj12001_rebas.py
from sys import stdin
input = stdin.readline

n, m = map(int, input().split())
x, y = [[] for _ in range(n)], [[] for _ in range(n)]
for i in range(n):
    x[i], y[i] = map(int, input().split())

def solve():
    ans = 1e9
    for i in range(n):
        for j in range(n):
            a, b, c, d = [0]*4
            for k in range(n):
                if x[k] < x[i]+1 and y[k] < y[j]+1:
                    a += 1
                if x[k] < x[i]+1 and y[k] > y[j]+1:
                    b += 1
                if x[k] > x[i]+1 and y[k] > y[j]+1:
                    c += 1
                if x[k] > x[i]+1 and y[k] < y[j]+1:
                    d += 1
            gap = max(a, b, c, d)
            ans = min(ans, gap)
    print(ans)

solve()

File: Algorithm and Data Structure/Baekjoon/12001번 Load Balancing (Bronze)/boj12001_rebas.cc
#include <cstdio>
#include <vector>
#include <algorithm>
using namespace std;

int n, m;
vector<int> x, y;

void solve() {
    int ans = 1e9;
    for (int i=0; i<n; i++) {
        for (int j=0; j<n; j++) {
            int a = 0, b = 0, c = 0, d = 0;
            for (int k=0; k<n; k++) {
                if (x[k] < x[i]+1 && y[k] < y[j]+1) a++;
                if (x[k] < x[i]+1 && y[k] > y[j]+1) b++;
                if (x[k] > x[i]+1 && y[k] < y[j]+1) c++;
                if (x[k] > x[i]+1 && y[k] > y[j]+1) d++;
            }
            int cow = max(max(max(a, b), c), d);
            if (ans > cow) ans = cow;
        }
    }
    printf("%d\n", ans);
}

int main() {
    scanf("%d %d", &n, &m);
    for (int i=0; i<n; i++) {
        int a, b;
        scanf("%d %d", &a, &b);
        x.push_back(a);
        y.push_back(b);
    }
    solve();
    return 0;
}


